packages/ember-glimmer/lib/syntax/curly-component.js

```diff
@@ -1,15 +1,41 @@
-import { StatementSyntax, ValueReference } from 'glimmer-runtime';
+import { StatementSyntax, ValueReference, EvaluatedArgs, EvaluatedNamedArgs, EvaluatedPositionalArgs } from 'glimmer-runtime';
 import { TO_ROOT_REFERENCE, AttributeBindingReference, applyClassNameBinding } from '../utils/references';
 import { DIRTY_TAG, IS_DISPATCHING_ATTRS, HAS_BLOCK } from '../component';
 import { assert } from 'ember-metal/debug';
 import processArgs from '../utils/process-args';
 import { privatize as P } from 'container/registry';
+import assign from 'ember-metal/assign';
 import get from 'ember-metal/property_get';
 import { ComponentDefinition } from 'glimmer-runtime';
 import Component from '../component';
 
 const DEFAULT_LAYOUT = P`template:components/-default`;
 
+export function validatePositionalParameters(named, positional, positionalParamsDefinition) {
```

Comment:
If all we are doing is `assert`ing (which gets removed from prod builds, I think?), should we just wrap the entire thing in `runInDebug`?